runtimes/base/src/xcm_config.rs

```diff
@@ -15,33 +15,67 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 use super::{
-	AccountId, AllPalletsWithSystem, Balances, EnsureRoot, ParachainInfo, ParachainSystem, PolkadotXcm, Runtime,
-	RuntimeCall, RuntimeEvent, RuntimeOrigin, WeightToFee, XcmpQueue,
+	AccountId, AllPalletsWithSystem, AssetId as AssetIdPalletAssets, Balance, Balances, EnsureRoot, ForeignAssets,
+	ParachainInfo, ParachainSystem, PolkadotXcm, Runtime, RuntimeCall, RuntimeEvent, RuntimeOrigin, WeightToFee,
+	XcmpQueue,
 };
+use core::marker::PhantomData;
 use frame_support::{
-	match_types, parameter_types,
-	traits::{ConstU32, Everything, Nothing},
+	ensure, match_types, parameter_types,
+	traits::{ConstU32, Contains, ContainsPair, Everything, Nothing, ProcessMessageError},
 	weights::Weight,
 };
 use pallet_xcm::XcmPassthrough;
+use polimec_xcm_executor::{
+	polimec_traits::{JustTry, Properties, ShouldExecute},
+	XcmExecutor,
+};
 use polkadot_parachain::primitives::Sibling;
 use polkadot_runtime_common::impls::ToAuthor;
+use sp_runtime::traits::MaybeEquivalence;
 use xcm::latest::prelude::*;
 use xcm_builder::{
 	AccountId32Aliases, AllowExplicitUnpaidExecutionFrom, AllowKnownQueryResponses, AllowSubscriptionsFrom,
-	AllowTopLevelPaidExecutionFrom, CurrencyAdapter, DenyReserveTransferToRelayChain, DenyThenTry, EnsureXcmOrigin,
-	FixedWeightBounds, IsConcrete, ParentIsPreset, RelayChainAsNative, SiblingParachainAsNative,
+	AllowTopLevelPaidExecutionFrom, CreateMatcher, CurrencyAdapter, DenyReserveTransferToRelayChain, DenyThenTry,
+	EnsureXcmOrigin, FixedRateOfFungible, FixedWeightBounds, FungiblesAdapter, IsConcrete, MatchXcm,
+	MatchedConvertedConcreteId, MintLocation, NoChecking, ParentIsPreset, RelayChainAsNative, SiblingParachainAsNative,
 	SiblingParachainConvertsVia, SignedAccountId32AsNative, SignedToAccountId32, SovereignSignedViaLocation,
 	TakeWeightCredit, UsingComponents, WithComputedOrigin,
 };
-use xcm_executor::XcmExecutor;
+// TODO: Check these numbers
+const DOT_ASSET_ID: AssetId = Concrete(RelayLocation::get());
+const DOT_PER_SECOND_EXECUTION: u128 = 0_0_000_001_000; // 0.0000001 DOT per second of execution time
+const DOT_PER_MB_PROOF: u128 = 0_0_000_001_000; // 0.0000001 DOT per Megabyte of proof size
```

Comment:
This is a good question. We definitely have to check these numbers. Question also is if we would like to allow free execution to users transferring assets from AssetHub to Polimec. I used the Barrier configured in the Testnet which allows explicit free execution from the AssetHub location. Which means that messages starting with UnpaidExecution received from AssetHub do not require any weight buying. These numbers would still be used for the transfer back, but for receiving assets, they wouldn't matter. 

So question is? How much are we willing to charge for Asset Transfers

Reply:
Is the free execution only for reserve transfers?
We could write a pre-compile macro (i.e proc macro) that does an http request to get the price of PLMC->USD, DOT->USD, assume USDC and USDT == USD, and try to charge the same USD amount we do for PLMC with the other asset.

I don't think it's worth to do that before MVP though.
For now any value within the same order of magnitude as PLMC is good enough, which I believe is what we have atm

Reply:
We shouldn't allow free transfers to anywhere. 
The reason we allow unpaid execution from AH is because the AH already charges for execution over there, preventing spam on both our networks.
If we do free execution on our side then we'll be open to spam